### PR TITLE
Make MYSQL_USER, MYSQL_PASSWORD and MYSQL_DATABASE optional

### PR DESCRIPTION
**Pull request by hhorak**
Implements https://github.com/openshift/mysql/issues/96

### CODE REVIEW

File: 5.5/contrib/common.sh [hunk 1]
```diff
@@ -20,13 +20,13 @@ mysql_password_regex='^[a-zA-Z0-9_~!@#$%^&*()-=<>,.?;:|]+$'
...
   [ $# == 2 ] && echo "error: $1"
-  echo "You must specify following environment variables:"
+  echo "You must either specify the following environment variables:"
   echo "  MYSQL_USER (regex: '$mysql_identifier_regex')"
   echo "  MYSQL_PASSWORD (regex: '$mysql_password_regex')"
   echo "  MYSQL_DATABASE (regex: '$mysql_identifier_regex')"
-  echo "Optional:"
+  echo "Or the following environment variable:"
   echo "  MYSQL_ROOT_PASSWORD (regex: '$mysql_password_regex')"
-  echo "Settings:"
```
Comment (bparees) [inline 5.5/contrib/common.sh]: "Or both"

**Comment by bparees** [comment]
@mnagy please review.

File: 5.5/contrib/common.sh (hunk 1 above)
Comment (hhorak) [inline 5.5/contrib/common.sh]: Do you mean it this way?
```
...
Or the following environment variable:
  MYSQL_ROOT_PASSWORD (regex: '...')"
Or both.
Settings:
...
```

Comment (bparees) [inline 5.5/contrib/common.sh]: yes.      just trying to clarify to users that they can set both a ROOT password and a USER password, they aren't mutually exclusive, but at least one is required.

Comment (hhorak) [inline 5.5/contrib/common.sh]: Makes sense, fixed..

File: 5.6/test/run [hunk 2]
```diff
@@ -198,6 +197,7 @@ function try_image_invalid_combinations() {
 
 function run_container_creation_tests() {
   echo "  Testing image entrypoint usage"
+  assert_container_creation_fails "$@"
```
Comment (mnagy) [inline 5.6/test/run]: We never call run_container_creation_tests with arguments, so `"$@"` here is redundant.

Comment (hhorak) [inline 5.6/test/run]: Thanks, fixed..

**Comment by mnagy** [comment]
This is looking good. I'm going to run some more tests later today/early tomorrow and will lgtm if they pass.

**Comment by mnagy** [comment]
Tests have passed, LGTM.

**Pull request merged**